pageserver/src/tenant/mgr.rs

```diff
@@ -949,6 +949,19 @@ impl TenantManager {
                 (LocationMode::Attached(attach_conf), Some(TenantSlot::Attached(tenant))) => {
                     match attach_conf.generation.cmp(&tenant.generation) {
                         Ordering::Equal => {
+                            if attach_conf.attach_mode == AttachmentMode::Single {
+                                // We need to wait for LSN lease duration after transitioning into `AttachedSingle`
+                                // before doing any gc so that all previously granted lease will expire.
```

Comment:
the sentence is a bit off here, I think you meant to write "previously granted leases will have a chance to renew"?

Reply:
8bcaa47a311c418984e5d84f9ddc376065b602bc